src/com/google/javascript/jscomp/Compiler.java

```diff
@@ -668,6 +668,8 @@ private void compileInternal() {
     if (!precheck()) {
       return;
     }
+    
+    alwaysPreChecks(); // even in whitespace only mode
 
     if (!options.skipNonTranspilationPasses || options.lowerFromEs6()) {
       check();
```

Comment:
Yes, I saw that. However I am minimising passes and AST changes to be as
fast as possible, and to make debugging as close to original code as
possible without sourcemaps (I don't thinks these are supported unless
doing advanced compilation, or at least this was the case a while back).
And now that Chrome Canary supports all the ES6 killer features we use, we
don't use transpilation, hence none of the checks run.

Ideally there wouldn't be the lack of clarity as to which checks run.
Perhaps keep WS only mode as pure no-checks, and have another mode which
does the minimum needed for latest Chrome Canary build (I.e. concat only,
keeping comments/ws)?

Incidentally how practical is it to putting sourcemaps support in to all
comp levels (unless it already works), and to chaining sourcemaps I.e.
providing input sourcemaps for previous external transformations? I'm
looking at adding a JSX transpilation pass (initially using Babel in
Nashorn and it would emit sourcemaps - adapting your parser to recognise
JSX is a bit beyond my abilities).
On 27 Aug 2015 6:57 pm, "Ben Lickly" <notifications@github.com> wrote:

> In src/com/google/javascript/jscomp/Compiler.java
> <https://github.com/google/closure-compiler/pull/1094#discussion_r38118848>
> :
>
> > @@ -668,6 +668,8 @@ private void compileInternal() {
> >      if (!precheck()) {
> >        return;
> >      }
> > +
> > +    alwaysPreChecks(); // even in whitespace only mode
> >
> >      if (!options.skipNonTranspilationPasses || options.lowerFromEs6()) {
> >        check();
>
> We actually already have a case where whitespace only mode runs the checks
> (when we are transpiling from ES6 down to ES3/5), so it's reasonable to
> just run check unconditionally, and add this logic to the beginning of
> getChecks in DefaultPassConfig.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/google/closure-compiler/pull/1094/files#r38118848>.
>